src/main/java/com/zaxxer/nuprocess/osx/ProcessKqueue.java

```diff
@@ -33,13 +36,12 @@
  */
 final class ProcessKqueue extends BaseEventProcessor<OsxProcess>
 {
-   private static final int KEVENT_POOL_SIZE = 16;
+   private static final int NUM_KEVENTS = 64;
    private static final TimeSpec timeSpec;
    private static final int JAVA_PID;
 
    private volatile int kqueue;
-   private Kevent triggeredEvent;
-   private BlockingQueue<Kevent> keventPool;
+   private ThreadLocal<KeventArray> keventArrays;
```

Comment:
I have had no end of troubles with ThreadLocals and application containers, like Tomcat, that sniff for ThreadLocal leaks...

https://github.com/brettwooldridge/HikariCP/issues?utf8=✓&q=is%3Aissue+is%3Aclosed+threadlocal+

[Classloader Leaks](http://www.slideshare.net/mjiderhamn/classloader-leaks/28)